src/common/init.cpp

```diff
@@ -61,7 +63,48 @@ std::optional<ConfigError> InitConfig(ArgsManager& args, SettingsAbortFn setting
         if (!fs::exists(net_path)) {
             fs::create_directories(net_path / "wallets");
         }
+#ifdef __APPLE__
+        struct PathCheck {
+            fs::path path;
+            std::string_view description;
+        };
 
+        std::array<PathCheck, 2> paths{{
+            {args.GetDataDirNet(), "data directory"},
+            {args.GetBlocksDirPath(), "blocks directory"}
+        }};
+
+        std::vector<std::string> exfat_paths;
+        std::vector<std::string> error_paths;
+
+        for (const auto& check : paths) {
+            FSType fs_type = GetFilesystemType(check.path);
+            switch(fs_type) {
+                case FSType::EXFAT:
+                    exfat_paths.push_back(strprintf("%s (\"%s\")",
+                        check.description,
+                        fs::PathToString(check.path)));
+                    break;
+                case FSType::ERROR:
+                    error_paths.push_back(strprintf("%s (\"%s\")",
+                        check.description,
+                        fs::PathToString(check.path)));
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (!exfat_paths.empty()) {
+            InitWarning(strprintf(_("The following paths are on exFAT which is known to have intermittent corruption problems on MacOS: %s. "
```

Comment:
small nit:
```suggestion
            InitWarning(strprintf(_("The following paths are on exFAT which is known to have intermittent corruption problems on macOS: %s. "
```

Reply:
MacOS corrected to macOS in according to #31453 (comment)

Reply:
Sorry, my bad here, I should've checked properly. That's correct, Apple [rebranded](https://en.wikipedia.org/wiki/macOS) MacOS to macOS in 2016.